packages/modules/viewer-react/src/components/iModel/IModelLoader.tsx

```diff
@@ -104,150 +92,78 @@ const Loader: React.FC<ModelLoaderProps> = React.memo(
         onIModelConnected(imodelConnection);
       }
       setConnection(imodelConnection);
+      return imodelConnection;
     };
 
-    const getViewState = useCallback(async () => {
-      if (!connection || (!connection.isBlank && connection.isClosed)) {
-        setViewState(undefined);
-        return;
+    const getModelConnection = useCallback(async (): Promise<
+      IModelConnection | undefined
+    > => {
+      if (blankConnection) {
```

Comment:
should you check that its mounted too?

Reply:
this extra function was unnecessary. I removed it